packages/taquito/src/contract/rpc-contract-provider.ts

```diff
@@ -585,6 +588,24 @@ export class RpcContractProvider
     );
   }
 
+  async ballot(params: BallotParams) {
+    const publicKeyHash = await this.signer.publicKeyHash();
+
+    if (params.source && validateAddress(params.source) !== ValidationResult.VALID) {
+      throw new InvalidAddressError(params.source);
+    }
+    const source = params.source ?? publicKeyHash;
+    const operation = await createBallotOperation({
+      ...params,
+      source: source,
+    });
+    const ops = await this.addRevealOperationIfNeeded(operation, publicKeyHash);
```

Comment:
Does the source really need to be revealed for a ballot operation?

Reply:
good point, I don't think we will need to